docs/service-mesh/10-06-sidecar-version.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Istio sidecar version after Kyma upgrade
+type: Troubleshooting
+---
+
+
+Kyma has sidecar injection enabled by default - a sidecar is injected to every Deployment in a cluster, without the need to add any labels. For more information, read [this document](#details-sidecar-proxy-injection).
+
+The sidecar version in Pods must match the installed Istio version, otherwise mesh connectivity may be broken.
+This is not an issue when installing Kyma, but it may be a problem during upgrades. When Kyma is upgraded to a new version along with a new Istio version, existing sidecars injected into Pods remain in an original version.
+Kyma contains an `istio-proxy-reset` job, that performs a rollout for most common workload types like deployments, daemonsets, etc. The job ensures all Kyma components are properly updated.
+Certain kinds of user-defined workloads can't be rolled out automatically, for example, a standalone Pod without any backing management mechanism (like ReplicaSet or a Job).
+Such user-defined workloads (that are not part of Kyma) must be manually restarted to work correctly with the updated Istio version.
+
+To find if any pods/workloads require a manual restart, you can:
+
+* Get the list of Pods with outdated sidecar version. You can get the Istio version for a Kyma release from the Istio chart version located in `resources/Istio/Chart.yaml` file. The returned list is in `name/namespace` format, empty output means no Pods require migration. Use the following command:
```

Comment:
Maybe you could provide a command for getting Istio version? For example from istiod deployment and export it.

Reply:
Added